posts/ekp1k80/extension_chapter_2.md

```diff
@@ -0,0 +1,273 @@
+}
+```
+
+***
+
+Modify Popup layout size `src/popup/index.css`
+
+```
+html {
+  width: 720px;
+  height: 480px;
+}
+body {
+  margin: 0;
+  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
+    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
+    sans-serif;
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
+  width: 100%;
+  height: 100%;
+  padding: 2%;
+}
+
+code {
+  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
+    monospace;
+}
+```
+
+After that build it
+
+```
+yarn build
+```
+And with that you are gonna have your extension configured with React
+![extension_chapter_2_video1.gif](./assets/extension_chapter_2_video1.gif)
+
+## Explanation
+So... my head is bumping, i spend more than 3 hours writing this and i know you are gonna read it in 5 minutes or less, F*** me.
+
+So, what we just did?
+
+### What is an Extension
+An extension is just a folder with a manifest.json file and some more stuff, but the basic things is the manifest.json.
+
+folder/
+└── logo.png
+└── index.html
+└── manifest.json
+
+manifest.json
+```
+{
+  "manifest_version": 3,
+  "name": "Hello Extensions",
+  "description": "Base Level Extension",
+  "version": "1.0",
+  "action": {
+    "default_popup": "index.html",
+    "default_icon": "logo.png"
+  }
+}
+```
+
+index.html
+```
+<html>
+  <body>
+    <h1>Hello Extensions</h1>
+  </body>
+</html>
+```
+***
+
+
+After installed, you have your extension
+![extension_chapter_2_img1.png](./assets/extension_chapter_2_img1.png)
+
+But you want to create a react extension. So what you did was:
+1. Change your React manifest.json to be valid as an extension
+2. Move your CRA content to Popup
+3. Create content and backend for the extension
+4. Build everything separatly with customize-cra and react-app-rewired
+5. Build and run
+6. Also and PD. You install webextension-polyfill and some types to make typescript able to use Chrome API
+
+### What is popup, content and backend and why build everything separtly?
+1. The popup that you see when you click the icon in the toolbar.
+2. The content scripts that run on top of an existing website.
+3. The background scripts that run in the background of Google Chrome.
+
+***
+#### Popup
+This is a webpage you can create and is going to run when user click in the extension icon in the toolbar, you can put js, css, html, and more types of files. With the popup you can interact with the chrome API also.
+
+***
+#### Backend
+This is a script that is going to be running from the moment you install the extension, and you have all the control over it, it can comunicate with the popup and the content.
+
+***
+
+#### Content
+This is also an script but in this case is isolated from the backend and the popup, in this case is the only one who is going to be able to run and interact with the web application running in the tab. In is limited but it can interact with the backend.

```

Comment:
```suggestion
This is also a script but in this case it's isolated from the backend and the popup, it's the only one who is going to be able to run and interact with the web application running in the tab. It is limited but it can interact with the backend.

```